### PR TITLE
Add preliminary support for Cubic Bézier curve edge

### PR DESCRIPTION
**Pull request by kaisalmen**
This PR introduces preliminary support for cubic Bézier curve edges.

What has been achieved:
* [x] Support one or many Bezier curves
* [x] Allow interactive manipulation of curves
* [x] Interactively add and remove new curves on an edge
* [x] Smooth interpolation on curve junctions

Issues that should be fixed before PR is ready
* [x] Add +/- to green/red circles for add/remove curve
* [x] New Curve should be created on "click" and not "move" of add/remove
* [x] Visual connections (lines) between control points and curve endings

What is still missing (reason for being preliminary). To be discussed if done in scope of this PR or later:
* [ ] Add unit tests
* [ ] Support Text-Layout
* [ ] Line jumps according curve

The classdiagram has been enhanced to show the new capabilities.

### CODE REVIEW

**Comment by kaisalmen** [comment]
Ready for review. The last open point is now resolved: Add MouseListener that reacts on mouseDown on bezier-add and bezier-remove handles. The event riggers a new Action and Command that adds or removes a curve segment.

**Review by spoenemann — APPROVED** [review]
Thanks! I have a few more remarks, but I'll merge it anyway so we can start the work on #243. @kaisalmen please create a new PR to address the remaining comments when the restructuring is done (will submit a PR for that today).

File: src/features/routing/bezier-edge-router.ts [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+    @inject(EdgeRouterRegistry) @optional() edgeRouterRegistry?: EdgeRouterRegistry;
+
+    mouseDown(target: SModelElement, event: MouseEvent): Action[] {
+        if (!(target instanceof SRoutingHandle) || (target.kind !== 'bezier-add' && target.kind !== 'bezier-remove')) {
+            return [];
+        }
+
+        const result = [];
+        let router;
+        const element = target.root.index.getById(target.id);
```
[the remark this answers is not in the document]
Comment (kaisalmen) [reply src/features/routing/bezier-edge-router.ts:329]: True. I will simplify it.

File: src/features/routing/bezier-edge-router.ts [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+}
+
+export class BezierCurveAction implements Action {
+    static readonly KIND = 'bezierAction';
+    readonly kind = BezierCurveAction.KIND;
+    pointIndex: number;
+    edgeId: string;
+    actionTask: BezierActionTask;
+
+    constructor(@inject(TYPES.IEdgeRouter) protected readonly router: BezierEdgeRouter,
```
[the remark this answers is not in the document]
Comment (kaisalmen) [reply src/features/routing/bezier-edge-router.ts:356]: Ok, the action will only contain the targetId and a value from the enum in the future.

File: src/features/routing/bezier-edge-router.ts [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+        actionTask: BezierActionTask, pointIndex: number, edgeId: string) {
+        this.actionTask = actionTask;
+        this.pointIndex = pointIndex;
+        this.edgeId = edgeId;
+    }
+
+    getRouter(): BezierEdgeRouter {
```
[the remark this answers is not in the document]
Comment (kaisalmen) [reply src/features/routing/bezier-edge-router.ts:363]: ✅ It was a bit late yesterday evening.

File: src/features/routing/bezier-edge-router.ts [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+        return this.pointIndex;
+    }
+
+    getEdgeId(): string {
+        return this.edgeId;
+    }
+};
+
+@injectable()
+export class BezierCurveCommand extends Command {
```
[the remark this answers is not in the document]
Comment (kaisalmen) [reply src/features/routing/bezier-edge-router.ts:377]: Yes, good idea. I will rename the action accordingly

**Pull request merged**